File: 2021_09_29/Pascal/1.pas
var ans : longint;
	x1, x2 : longint;

function digit_sum(n : longint) : longint;
Begin
	digit_sum := n div 100 + (n div 10) mod 10 + n mod 10;
End;

BEGIN
	ans := 0;
	for x1 := 100 to 999 do
		for x2 := 100 to 999 do
			if (x1 mod digit_sum(x2) = 0) and (x2 mod digit_sum(x1) = 0) then
				Inc(ans);
	Writeln(ans);
END.

File: 2021_09_29/C++/1.cpp
#include <iostream>

int digit_sum(int n) {
	return n / 100 + (n / 10) % 10 + n % 10;
}

int main() {
	int ans = 0;
	for (int x1 = 100; x1 < 1000; x1++)
		for (int x2 = 100; x2 < 1000; x2++)
			if (x1 % digit_sum(x2) == 0 && x2 % digit_sum(x1) == 0)
				ans++;
	std::cout << ans << std::endl;
	return 0;
}
